### PR TITLE
Desktop: Put all keybindings in default

### LINKED ISSUE #3499: Undo and Select All not working in Code Mirror editor
- macOS, version 1.0.229

@CalebJohn, I wonder if Undo and Select All ever worked in Code Mirror? Or is it something I broke with my recent refactoring of the command system?

### PR DESCRIPTION
**Pull request by CalebJohn**
This is an attempt to fix #3499 , i think the issue is that `keymap.macosDefault` is just a reference and the actually used map is default.

### CODE REVIEW

**Comment by laurent22** [comment]
Doesn't it mean that both regular and macOS shortcuts will be defined at the same time? i.e. Cmd+A and Ctrl+A do the same thing? If so I don't think that's the right approach as we should aim to be very specific about what shortcuts we support, so it should be Cmd+A only on macOS and Ctrl+A only on all other platforms.

Isn't it possible to get it working by using [addKeyMap](https://codemirror.net/doc/manual.html#addKeyMap) in combination with the [keyMap option](https://codemirror.net/doc/manual.html#option_keyMap)? Perhaps you could even set the keyMap option to an empty array, then define explicitely all the kemaps we support via addKeyMap. That way it would be easier to have a conditional for macOS/Windows/Linux. It's true that the CodeMirror doc is not the clearest for this though.

**Comment by CalebJohn** [comment]
`addKeyMap` is a method that allows you to change keymaps after the editor has been initialized (which is useful for plugins), in our case it's unnecessary because we have full control over the editor lifetime. 

What I did here is that I manually set the default based on `shim.isMac()` so the mac version only ever gets mac commands. That way the keyMap option can remain (emacs, vim, default). But macOs only sees the mac default and windows/linux sees the others.

File: ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/Editor.tsx [hunk 1]
```diff
@@ -105,55 +106,62 @@ function Editor(props: EditorProps, ref: any) {
...
-		'Cmd-]': 'indentMore',
-		'Cmd-/': 'toggleComment',
-		'Cmd-Opt-S': 'sortSelectedLines',
-		'Opt-Up': 'swapLineUp',
-		'Opt-Down': 'swapLineDown',
-		'fallthrough': 'basic',
-	};
+
+	if (shim.isMac()) {
+		CodeMirror.keymap.default = {
```
Comment (laurent22) [inline ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/Editor.tsx:111]: Should be `keyMap`

Comment (CalebJohn) [reply ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/Editor.tsx]: thanks!

**Comment by CalebJohn** [comment]
Good suggestion! My thought was that since it's stable it didn't need to be in an effect. This was of course wrong, it's fixed now.

**Comment by laurent22** [comment]
Ok, there was still a typo keymap/keyMap which was crashing the app on macOS, but once fixed it's working fine. Thanks for the update!

**Pull request merged**